backend/api.test/EventHandlers/TestMissionEventHandler.cs

```diff
@@ -381,5 +382,71 @@ public async void ReturnHomeMissionNotScheduledIfRobotIsNotLocalized()
             Assert.False(await _missionRunService.PendingOrOngoingReturnToHomeMissionRunExists(robot.Id));
 
         }
+
+        [Fact]
+        public async void ReturnHomeMissionCancelledIfNewMissionScheduled()
+        {
+            // Arrange
+            var installation = await _databaseUtilities.NewInstallation();
+            var plant = await _databaseUtilities.NewPlant(installation.InstallationCode);
+            var deck = await _databaseUtilities.NewDeck(installation.InstallationCode, plant.PlantCode);
+            var area = await _databaseUtilities.NewArea(installation.InstallationCode, plant.PlantCode, deck.Name);
+            var robot = await _databaseUtilities.NewRobot(RobotStatus.Busy, installation, area);
+            var returnToHomeMission = await _databaseUtilities.NewMissionRun(installation.InstallationCode, robot, area, true, MissionRunType.ReturnHome, MissionStatus.Ongoing, Guid.NewGuid().ToString());
+            var missionRun = await _databaseUtilities.NewMissionRun(installation.InstallationCode, robot, area, true, MissionRunType.Normal, MissionStatus.Pending, Guid.NewGuid().ToString());
+
+            Thread.Sleep(100);
+
+            // Act
+            var eventArgs = new MissionRunCreatedEventArgs(missionRun.Id);
+            _missionRunService.RaiseEvent(nameof(MissionRunService.MissionRunCreated), eventArgs);
+
+            Thread.Sleep(500);
+
+            // Assert 
+            var updatedReturnHomeMission = await _missionRunService.ReadById(returnToHomeMission.Id);
+            Assert.True(updatedReturnHomeMission?.Status.Equals(MissionStatus.Cancelled));
+            Assert.True(updatedReturnHomeMission?.Tasks.FirstOrDefault()?.Status.Equals(TaskStatus.Cancelled));
+
+            // Act
+            // State we get after stopping mission in Isar
+            robot.Status = RobotStatus.Available;
+            await _robotService.Update(robot);
+            var robotAvailableArgs = new RobotAvailableEventArgs(robot.Id);
+            _missionSchedulingService.RaiseEvent(nameof(MissionSchedulingService.RobotAvailable), robotAvailableArgs);
```

Comment:
If we trigger a robot status event from the mqtt service it would set it to available and trigger this. Then we would not need the status update above